byzcoin/service.go

```diff
@@ -391,8 +393,16 @@ func (s *Service) AddTransaction(req *AddTxRequest) (*AddTxResponse, error) {
 		for found := false; !found; {
 			select {
 			case success := <-ch:
+				errMsg, exists := s.txErrorBuf.get(string(ctxHash))
 				if !success {
-					return nil, errors.New("transaction is in block, but got refused")
+					if !exists {
+						return nil, errors.New("transaction is in block, but got refused for unknown error")
+					}
+					return nil, errors.New(trimErrorMsg(errMsg))
+				}
+
+				if exists {
+					log.Warn(s.ServerIdentity(), "transaction is accepted but there are errors: ", errMsg)
```

Comment:
```suggestion
					log.Warn(s.ServerIdentity(), "transaction is in block but got refused for the following reason: ", errMsg)
```

Reply:
i wanted to say the transaction is accepted because the `success` flag is true, but for some reason there is a an error message (which should never happen) so we print a warning